crates/primitives/src/precompile.rs

```diff
@@ -1,10 +1,13 @@
+use crate::Env;
 use alloc::vec::Vec;
 
 /// A precompile operation result.
+///
+/// Returns either `Ok((gas_used, return_bytes))` or `Err(error)`.
 pub type PrecompileResult = Result<(u64, Vec<u8>), PrecompileError>;
 
 pub type StandardPrecompileFn = fn(&[u8], u64) -> PrecompileResult;
-pub type CustomPrecompileFn = fn(&[u8], u64) -> PrecompileResult;
+pub type EnvPrecompileFn = fn(&[u8], u64, env: &Env) -> PrecompileResult;
```

Comment:
any reason for this rename? i assume it would be a breaking change, i'm not sure if this pr is supposed to be breaking in general?

Reply:
Mostly as it was not used, it was a idea to add "Custom" precompiles and allows an external user to add their precompiles, but usage was lacking, should be fine to break it as the signature is the same as a `StandardPrecompileFn`.